api/api.go

```diff
@@ -72,18 +72,17 @@ func NewBlockChainAPI(
 // wasn't synced up to a block where EIP-155 is enabled, but this behavior caused issues
 // in CL clients.
 func (b *BlockChainAPI) ChainId() *hexutil.Big {
-	return (*hexutil.Big)(b.config.ChainID)
+	return (*hexutil.Big)(b.config.EVMNetworkID)
 }
 
 // BlockNumber returns the block number of the chain head.
-func (b *BlockChainAPI) BlockNumber() hexutil.Uint64 {
+func (b *BlockChainAPI) BlockNumber() (hexutil.Uint64, error) {
```

Comment:
I believe the way it works (and I might be wrong but from testing I've noticed this) is that there is some ethereum server implementation that checks if there's an error returned from the handler it will only return an error response, something containing a code and description etc, but if there's no error it will only return the value, so it doesn't affect the return value if no error is present. Then web3 clients handle those errors accordingly. The reason geth doesn't have this errors is because their functions in some API handlers don't produce any errors. But for those who can they return an error. So it's natural for us to have different return values.